crates/zk-por-core/src/bin/prover.rs

```diff
@@ -0,0 +1,154 @@
+}
+
+fn main() {
+    let cfg = ProverConfig::try_new().unwrap();
+    let trace_cfg: TraceConfig = cfg.log.into();
+    init_tracing(trace_cfg);
+
+    const RECURSION_FACTOR: usize = 64;
+    const BATCH_PROVING_THREADS_NUM: usize = 4;
+    const RECURSIVE_PROVING_THREADS_NUM: usize = 2;
+
+    if cfg.prover.hyper_tree_size as usize != RECURSION_FACTOR {
+        panic!("The hyper_tree_size is not configured to be equal to 64 (Recursion_Factor)");
+    }
+    let batch_size = cfg.prover.batch_size as usize;
+    let asset_num = cfg.prover.num_of_tokens as usize;
+
+    // the path to dump the final generated proof
+    let mut bench_mode = true;
+    let args: Vec<String> = env::args().collect();
+    let arg1 = args.get(1).expect(
+        "Please provide the first argument, either proof path or '--bench' for benchmark mode",
+    );
+    let mut account_reader: Box<dyn AccountParser>;
+    if arg1 == "--bench" {
+        bench_mode = true;
+        let account_num = args
+            .get(2)
+            .expect("Please provide the account number as the second argument in benchmark mode")
+            .parse::<usize>()
+            .expect("The provided account number must be a valid usize");
+
+        if account_num % batch_size != 0 {
+            panic!("The account number must be a multiple of batch size");
+        }
+        account_reader = Box::new(parser::RandomAccountParser::new(account_num, asset_num));
+    } else {
+        let parser = FilesParser::new(FilesCfg {
+            dir: std::path::PathBuf::from_str(&cfg.prover.user_data_path).unwrap(),
+            batch_size: cfg.prover.batch_size,
+            num_of_tokens: cfg.prover.num_of_tokens,
+        });
+        parser.log_state();
+        account_reader = Box::new(parser);
+    }
+
+    // Hardcode three levels of recursive circuits, each branching out 64 children, with the last level with zk enabled.
+    // Hence given batch_size=1024, the current setting can support 268M (1024*64^3) accounts, enough for the foreseeable future. (Currently we have 10M accounts)
+    let recursive_circuit_configs = vec![STANDARD_CONFIG, STANDARD_CONFIG, STANDARD_ZK_CONFIG];
+
+    let circuit_registry = CircuitRegistry::<RECURSION_FACTOR>::init(
+        batch_size,
+        asset_num,
+        STANDARD_CONFIG,
+        recursive_circuit_configs,
+    );
+
+    tracing::info!(
+        "start to prove {} accounts with {} tokens, {} batch size",
+        account_reader.total_num_of_users(),
+        asset_num,
+        batch_size
+    );
+
+    let start = std::time::Instant::now();
+    let mut offset = 0;
+    let per_parse_account_num = 10 * batch_size;
+    let mut parse_num = 0;
+    let mut batch_proofs = vec![];
+    while offset < account_reader.total_num_of_users() {
+        parse_num += 1;
+        let accounts = account_reader.read_n_accounts(offset, per_parse_account_num);
+        let account_num = accounts.len();
+        assert_eq!(account_num % batch_size, 0);
```

Comment:
it is possible the last batch might not be a multiple of batch_size in actual case